### ISSUE TITLE
Problem with curl script on back-end

### ISSUE THREAD

**Issue by CoreyFedde** [ga-wdi-boston/capstone-project#798]
I am getting an error when I run a curl script to create a resource called "loan" on my database. The script I am running should create a loan with a reference to the User that creates it, but I am getting an error that says {"user":["must exist"]}~/wdi/projects/capstone/capstone-backend (users).

I've double checked to make sure I am signed in and the token I am passing the loan is valid. I think I am missing something in my back-end that in the authentication, but I've been unable to locate it.

Curl script
```ruby
TOKEN="BAhJIiVjNWE0NjViYjU3NmQ2NTJjZWEwZTA2ZGM2NThjNWE1ZAY6BkVG--afbcada9df63a5029c9e6d0dec0580a460e46694"
NAME="TEST"
LENDER="BOA"
PRINCIPAL="15000"
INTEREST="4.5"
LENGTH="4"
START="08-25-2017"
MIN="200"
CURRENT="200"

curl --include --request POST http://localhost:4741/loans \
  --header "Content-Type: application/json" \
  --header "Authorization: Token token=${TOKEN}" \
  --data '{
    "loans": {
      "name": "'"${NAME}"'",
      "lender": "'"${LENDER}"'",
      "principal": "'"${PRINCIPAL}"'",
      "interest_rate": "'"${INTEREST}"'",
      "loan_length": "'"${LENGTH}"'",
      "start_date": "'"${START}"'",
      "minimum_monthly_payment": "'"${MIN}"'",
      "current_monthly_payment": "'"${CURRENT}"'"
    }
  }'
```

**Comment by CoreyFedde** [comment]
This is the full error message: 

HTTP/1.1 422 Unprocessable Entity
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Content-Type: application/json; charset=utf-8
Cache-Control: no-cache
X-Request-Id: fa3aa1ba-03fc-4a56-b915-0a1583758559
X-Runtime: 0.009662
Vary: Origin
Transfer-Encoding: chunked

{"user":["must exist"]}~/wdi/projects/capstone/capstone-backend (users)

**Comment by jordanallain** [comment]
whats the controller action for creating a loan look like?

**Comment by CoreyFedde** [comment]
The unprocessable entity makes me think that I am missing a required field in the curl request. But looking at the schema, I haven't required any of the fields with null: false

```ruby
create_table "loans", force: :cascade do |t|
    t.string   "name"
    t.string   "lender"
    t.integer  "principal"
    t.integer  "interest_rate"
    t.integer  "loan_length"
    t.date     "start_date"
    t.integer  "minimum_monthly_payment"
    t.integer  "current_monthly_payment"
    t.datetime "created_at",              null: false
    t.datetime "updated_at",              null: false
    t.integer  "user_id"
    t.index ["user_id"], name: "index_loans_on_user_id", using: :btree
  end
```

**Comment by CoreyFedde** [comment]
Controller action is:
```ruby
def create
    @loan = Loan.new(loan_params)

    if @loan.save
      render json: @loan, status: :created, location: @loan
    else
      render json: @loan.errors, status: :unprocessable_entity
    end
  end
```

**Comment by jordanallain** [comment]
how is the user_id going to be assigned because you are not sending it in the request?

**Comment by jordanallain** [comment]
have you took a look at your full-stack project's create action?

**Comment by CoreyFedde** [comment]
Hi Jordan, 
thanks for the pointer. I realized I should probably be user current_user.loans.new. 
I've changed that, switched LoanController to inherit from the ProtectedController class. But I'm running into a new problem where I'm now getting back a Bad Request


HTTP/1.1 400 Bad Request
Content-Type: application/json; charset=utf-8
X-Request-Id: 659c227c-e107-43ff-854b-a9a4903610aa
X-Runtime: 0.072410
Vary: Origin
Content-Length: 11854

{"status":400,"error":"Bad Request","exception":"#\u003cActionController::ParameterMissing: param is missing or the value is empty: loan\u003e","traces":{"Application Trace":[{"id":1,"trace":"app/controllers/loans_controller.rb:49:in `loan_params'"}
